ext/soap/tests/bug71610.phpt

```diff
@@ -4,11 +4,20 @@ SOAP Bug #71610 - Type Confusion Vulnerability - SOAP / make_http_soap_request()
 soap
 --SKIPIF--
 <?php
-if (getenv("SKIP_ONLINE_TESTS")) die("skip online test");
+if (!file_exists(__DIR__ . "/../../../sapi/cli/tests/php_cli_server.inc")) {
+    echo "skip sapi/cli/tests/php_cli_server.inc required but not found";
+}
```

Comment:
I think that we should assume this file to always exist, and let the test fail if it does not. Otherwise it will silently become always-skipped if we move the file without updating it.

I would suggest to use `include __DIR__ . "/../../../sapi/cli/tests/skipif.inc"` instead.

Reply:
Well, for our purposes, at least regarding CI, this is certainly a good idea. However, some may build ext/soap via phpize (I assume distro managers do that), and in that case it is unlikely that the file exists. `include` will trigger a warning, and the test would be reported as borked.

Given that the file is already used elsewhere, that opcache has its own (diverged) copy, and that I would like to use this server for ext/standard/tests/http and ext/ftp/tests (these currently have servers which require the posix extension), I hope we can come up with a more general solution.

Not perfect, and might have issues, but what about storing symlinks in the repository?

Reply:
I didn't realize these tests could be executed in a standalone pecl build. Skipping the test like this seems reasonable then.

I agree that it would be nice to have a more general solution at some point.

Reply:
> Not perfect, and might have issues, but what about storing symlinks in the repository?

Do you mean symlinking the .inc file into ext/soap so that exports of the ext source code also contain it? I don't have experience with symlinks in git, but this could be an option indeed. An alternative could be to distribute/install a testing library along with run-tests.php or phpize (with the downside that evolving this library would be subject to BC).

Reply:
I've already did #16066, but @iluuu1994 suggested to go with a simple PHP include file instead of the symlinks (these have issues anyway; you need Windows 10 with developer mode enabled to have them properly work, and a somewhat recent git version, properly configure to accepts symlinks). And I guess, exporting the sources will not automatically resolve the symlinks. We could do that manually, though, through `./configure`. (not sure about the details)

> An alternative could be to distribute/install a testing library along with run-tests.php or phpize (with the downside that evolving this library would be subject to BC).

Yeah, that is what I would like, but I also see the problems.

Still, a first step to avoid those lengthy "parent paths" would be a good thing[TM].